### DISCUSSION TITLE
Do we store login history of users?

### DISCUSSION THREAD

**Discussion by stiwari99** [zitadel/zitadel#7699] [General]
Wondering if we store following information regarding user login
1. last login time (I am able to view this)
2. IP address 
3. location (user location from where login was initiated)

**Comment by dakshitha** [comment]
Regarding the last login time, you can use the the events API to filter for access token creation events.

Use the List Events endpoint to query for events related to access token creation. This endpoint can be specifically useful for auditing and monitoring purposes. This endpoint provides a detailed log of events. For the most recent login per user, you would need to process the data returned from the API to sort and filter based on the event timestamp.

https://zitadel.com/docs/apis/resources/admin/admin-service-list-events
https://zitadel.com/docs/guides/integrate/zitadel-apis/event-api#example-find-out-when-user-have-been-authenticated

**Reply by stiwari99** [reply]
Hi @dakshitha I have gone through API you mentioned but it does not have any way to get IP address and location of user from where login was requested
